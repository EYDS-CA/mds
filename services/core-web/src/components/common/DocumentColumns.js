import React from "react";
import * as Strings from "@common/constants/strings";
import { Button, Popconfirm } from "antd";
import { TRASHCAN } from "@/constants/assets";
import {
  renderDateColumn,
  renderDocumentLinkColumn,
  renderTextColumn,
} from "./CoreTableCommonColumns";

export const documentNameColumn = (
  documentNameColumnIndex = "document_name",
  title = "File Name"
) => {
  return renderDocumentLinkColumn(documentNameColumnIndex, title, true);
};

export const uploadDateColumn = (uploadDateIndex = "upload_date", title = "Uploaded") => {
  return renderDateColumn(uploadDateIndex, title, true, null, Strings.EMPTY_FIELD);
};

export const uploadedByColumn = (uploadedByIndex = "update_user", title = "Uploaded By") => {
  return renderTextColumn(uploadedByIndex, title);
};

export const removeFunctionColumn = (
  removeFunction,
  hideColumn = false,
  documentNameColumnIndex = "document_name",
  documentParent = ""
) => {
  const className = hideColumn ? "column-hide" : "test";
  return {
    key: "remove",
    className,
    render: (record) => (
      <div align="right" className>
        <Popconfirm
          placement="topLeft"
          title={`Are you sure you want to delete ${record[documentNameColumnIndex]}?`}
          onConfirm={(event) => removeFunction(event, record.key, documentParent)}
          okText="Delete"
          cancelText="Cancel"
        >
          <Button ghost type="primary" size="small">
            <img src={TRASHCAN} alt="remove" style={{ width: "24px" }} />
          </Button>
        </Popconfirm>
      </div>
    ),
  };
};
