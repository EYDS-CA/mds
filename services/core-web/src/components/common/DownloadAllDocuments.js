import React from "react";
import { notification } from "antd";
import PropTypes from "prop-types";
import { getDocumentDownloadToken } from "@common/utils/actionlessNetworkCalls";
import { DownloadOutlined } from "@ant-design/icons";

const propTypes = {
  submissions: PropTypes.arrayOf(PropTypes.any).isRequired,
};

const downloadDocument = (url) => {
  const a = document.createElement("a");
  a.href = url.url;
  a.download = url.filename;
  a.style.display = "none";
  document.body.append(a);
  a.click();
  a.remove();
};

const waitFor = (conditionFunction) => {
  const poll = (resolve) => {
    if (conditionFunction()) resolve();
    else setTimeout(() => poll(resolve), 400);
  };

  return new Promise(poll);
};

export const DownloadAllDocuments = (props) => {
  const handleDownloadAll = () => {
    const docURLS = [];

    const totalFiles = props.submissions.length;
    if (totalFiles === 0) {
      return;
    }

    props.submissions.forEach((doc) =>
      getDocumentDownloadToken(doc.documentManagerGuid, doc.filename, docURLS)
    );

    let currentFile = 0;
    waitFor(() => docURLS.length === props.submissions.length).then(async () => {
      // eslint-disable-next-line no-restricted-syntax
      for (const url of docURLS) {
        currentFile += 1;

        downloadDocument(url);

        // eslint-disable-next-line
        await new Promise((resolve) => setTimeout(resolve, 3000));
      }
      notification.success({
        message: `Successfully Downloaded: ${totalFiles} files.`,
        duration: 10,
      });
    });
  };
  return (
    <div className="custom-menu-item">
      <button type="button" className="full" onClick={() => handleDownloadAll()}>
        <DownloadOutlined style={{ color: "#5e46a1", paddingLeft: "8px", paddingRight: "22px" }} />
        Download All
      </button>
    </div>
  );
};

export default DownloadAllDocuments;
