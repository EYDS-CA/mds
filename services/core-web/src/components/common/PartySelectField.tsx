import React, { FC, useCallback, useEffect, useMemo, useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { debounce } from "lodash";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faHashtag } from "@fortawesome/pro-light-svg-icons";

import { Divider } from "antd";
import { MailOutlined, PhoneOutlined, PlusOutlined } from "@ant-design/icons";
import { Field } from "redux-form";
import { getSearchResults } from "@mds/common/redux/selectors/searchSelectors";
import { getLastCreatedParty } from "@mds/common/redux/selectors/partiesSelectors";
import { fetchSearchResults } from "@mds/common/redux/actionCreators/searchActionCreator";
import { setAddPartyFormState } from "@mds/common/redux/actionCreators/partiesActionCreator";
import { createItemIdsArray, createItemMap } from "@common/utils/helpers";
import { Validate } from "@common/utils/Validate";
import LinkButton from "@mds/common/components/common/LinkButton";
import RenderSelect from "@mds/common/components/forms/RenderSelect";

const renderAddPartyFooter = (showAddParty, partyLabel) => (
  <div className="wrapped-text">
    <Divider style={{ margin: "0" }} />
    <p className="footer-text">{`Can't find the ${partyLabel} you are looking for?`}</p>
    <LinkButton onClick={showAddParty}>
      <PlusOutlined className="padding-small--right" />
      {`Add a new ${partyLabel}`}
    </LinkButton>
  </div>
);

const transformData = (data, options, header) => {
  if (data.length === 0) {
    return [];
  }
  const transformedData = data
    .map((opt) => ({
      value: options[opt].party_guid,
      originalValue: options[opt],
      label: (
        <div>
          <span>{options[opt].name}</span>
          <div className="inline-flex">
            <div className="padding-right">
              <FontAwesomeIcon icon={faHashtag} />
            </div>
            <span>{options[opt].party_orgbook_entity?.registration_id}</span>
          </div>
          <div className="inline-flex">
            <div className="padding-right">
              <MailOutlined className="icon-xs" />
            </div>
            <span>
              {Validate.checkEmail(options[opt].email) ? options[opt].email : "Email Unknown"}
            </span>
          </div>
          <div className="inline-flex">
            <div className="padding-right">
              <PhoneOutlined className="icon-xs" />
            </div>
            <span>
              {options[opt].phone_no} {options[opt].phone_ext ? `x${options[opt].phone_ext}` : ""}
            </span>
          </div>
        </div>
      ),
    }))
    .filter((opt) => opt.value);

  // Display header only if desired (Add new party behavior is enabled.)
  if (header) {
    transformedData.unshift({ value: "header", label: header });
  }

  return transformedData;
};

interface PartySelectFieldProps {
  id?: string;
  name?: string;
  label?: string;
  partyLabel?: string;
  // set to true to only see people
  person?: boolean;
  // set to true to only see organizations
  organization?: boolean;
  allowAddingParties: boolean;
  validate: any[];
  searchResults?: any;
  initialValues?: any;
  allowNull?: boolean;
  disabled?: boolean;
  onSelect?: any;
}

export const PartySelectField: FC<PartySelectFieldProps> = ({
  initialValues,
  onSelect,
  person,
  organization,
  partyLabel = "contact",
  allowAddingParties,
  validate,
  disabled,
  allowNull,
  ...rest
}) => {
  const dispatch = useDispatch();

  const searchResults = useSelector(getSearchResults);
  const lastCreatedParty = useSelector(getLastCreatedParty);

  const [selectedOption, setSelectedOption] = useState({ value: "", label: "" });
  const [partyDataSource, setPartyDataSource] = useState([]);
  const [searching, setSearching] = useState(false);

  const handleFetchSearchResults = useCallback(
    (searchTerm: string, searchType: string) => {
      setSearching(true);
      dispatch(fetchSearchResults(searchTerm, searchType));
    },
    [dispatch]
  );

  const debouncedSearch = useMemo(
    () => debounce(handleFetchSearchResults, 1000),
    [handleFetchSearchResults]
  );

  useEffect(() => {
    if (searchResults) {
      setSearching(false);
    }
  }, [searchResults]);

  useEffect(() => {
    if (initialValues?.label) {
      handleSearch(initialValues.label);
      setSelectedOption({
        value: initialValues.value,
        label: initialValues.label,
      });
    }
  }, [initialValues]);

  const showAddPartyForm = () => {
    dispatch(
      setAddPartyFormState({
        showingAddPartyForm: true,
        person: !person,
        organization: !organization,
        partyLabel: partyLabel,
        initialValues,
      })
    );
  };

  useEffect(() => {
    if (searchResults || lastCreatedParty) {
      let filteredParties = searchResults?.party?.map((sr) => sr.result);

      if (filteredParties) {
        if (organization && !person) {
          filteredParties = filteredParties.filter(
            ({ party_type_code }) => party_type_code === "ORG"
          );
        } else if (person && !organization) {
          filteredParties = filteredParties.filter(
            ({ party_type_code }) => party_type_code === "PER"
          );
        }

        if (lastCreatedParty) {
          filteredParties.unshift(lastCreatedParty);
        }

        const newPartyDataSource = transformData(
          createItemIdsArray(filteredParties, "party_guid"),
          createItemMap(filteredParties, "party_guid"),
          allowAddingParties && renderAddPartyFooter(showAddPartyForm, partyLabel)
        );

        if (newPartyDataSource) {
          setPartyDataSource(newPartyDataSource);
        }
      }
    }

    if (lastCreatedParty) {
      setSelectedOption({
        value: lastCreatedParty.party_guid,
        label: lastCreatedParty.name,
      });
    }
  }, [lastCreatedParty, searchResults, organization, person, allowAddingParties, partyLabel]);

  const handleSearch = (value) => {
    if (value.length >= 2) {
      setSearching(true);
      debouncedSearch(value, "party");
    }
    setSelectedOption({ value, label: value });
  };

  const handleSelect = (value, option) => {
    setSelectedOption(option);
    if (onSelect) {
      onSelect({
        ...option,
        ...value.originalValue,
      });
    }
  };

  // Validator to ensure the selected option is in the collection of available options.
  // This validator is appened to any validators passed in from the form in the render function below.
  // eslint-disable-next-line consistent-return
  const validOption = (value) => {
    // ignore this validation if an initialValue is passed in
    if (initialValues && initialValues !== selectedOption && !allowNull) {
      return partyDataSource.find((opt) => opt.value === value)
        ? undefined
        : `Invalid ${partyLabel}`;
    }
  };

  return (
    <Field
      className="left"
      loading={searching}
      disabled={disabled}
      {...rest}
      component={RenderSelect}
      handleSearch={handleSearch}
      handleSelect={handleSelect}
      validate={validate.concat(validOption)}
      dataSource={partyDataSource}
      selectedOption={selectedOption}
    />
  );
};

export default PartySelectField;
