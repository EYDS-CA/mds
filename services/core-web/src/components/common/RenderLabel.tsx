import React, { FC } from "react";
import { Typography, Divider, Row, Col, Form } from "antd";
import { BaseInputProps } from "@mds/common/components/forms/BaseInput";


interface RenderLabelProps extends BaseInputProps {
  indentText?: string;
  className?: string;
}
/**
 * 
 * @deprecated - only used in GenerateDocumentFormField, don't add to new cases
 */
const RenderLabel: FC<RenderLabelProps> = ({ className = "", ...props }) => {
  return (
    <Form.Item>
      <Row className={className}>
        {props.indentText && (
          <>
            <Col span={2}>
              <Typography.Paragraph disabled>{props.indentText}</Typography.Paragraph>
            </Col>
            <Col span={1}>
              <Divider type="vertical" plain />
            </Col>
          </>
        )}
        <Col span={props.indentText ? 21 : 24}>
          <Typography.Paragraph
            ellipsis={{
              expandable: true,
              rows: 3,
            }}
          >
            {props.input.value}
          </Typography.Paragraph>
        </Col>
      </Row>
    </Form.Item>
  );
};

export default RenderLabel;
