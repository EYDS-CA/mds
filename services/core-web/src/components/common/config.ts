import RenderAutoComplete from "@mds/common/components/forms/RenderAutoComplete";
import RenderAutoSizeField from "@mds/common/components/forms/RenderAutoSizeField";
import RenderCheckbox from "@mds/common/components/forms/RenderCheckbox";
import RenderDate from "@mds/common/components/forms/RenderDate";
import RenderField from "@mds/common/components/forms/RenderField";
import RenderFileUpload from "@mds/common/components/forms/RenderFileUpload";
import RenderGroupCheckbox from "@mds/common/components/forms/RenderGroupCheckbox";
import RenderMultiSelect from "@mds/common/components/forms/RenderMultiSelect";
import RenderRadioButtons from "@mds/common/components/forms/RenderRadioButtons";
import RenderSelect from "@mds/common/components/forms/RenderSelect";
import RenderCascader from "@mds/common/components/forms/RenderCascader";
import RenderGroupedSelect from "@mds/common/components/forms/RenderGroupedSelect";
import RenderLargeSelect from "@mds/common/components/forms/RenderLargeSelect";
import RenderTime from "@mds/common/components/forms/RenderTime";
import RenderLabel from "./RenderLabel";
import RenderYear from "@mds/common/components/forms/RenderYear";

export const renderConfig = {
  CHECKBOX: RenderCheckbox,
  GROUP_CHECK_BOX: RenderGroupCheckbox,
  AUTOCOMPLETE: RenderAutoComplete,
  AUTO_SIZE_FIELD: RenderAutoSizeField,
  CASCADER: RenderCascader,
  DATE: RenderDate,
  TIME: RenderTime,
  YEAR: RenderYear,
  FIELD: RenderField,
  SCROLL_FIELD: RenderAutoSizeField,
  SELECT: RenderSelect,
  LARGE_SELECT: RenderLargeSelect,
  MULTI_SELECT: RenderMultiSelect,
  RADIO: RenderRadioButtons,
  GROUPED_SELECT: RenderGroupedSelect,
  LABEL: RenderLabel,
  FILE_UPLOAD: RenderFileUpload,
};
