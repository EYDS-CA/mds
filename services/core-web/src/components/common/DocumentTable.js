import React from "react";
import PropTypes from "prop-types";
import { Table, Popconfirm, Button } from "antd";
import { formatDate, dateSorter, nullableStringSorter } from "@common/utils/helpers";
import { some } from "lodash";
import DocumentLink from "@/components/common/DocumentLink";
import { TRASHCAN } from "@/constants/assets";
import CustomPropTypes from "@/customPropTypes";
import { closeModal, openModal } from "@common/actions/modalActions";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { modalConfig } from "@/components/modalContent/config";

const propTypes = {
  documents: PropTypes.arrayOf(CustomPropTypes.documentRecord),
  isViewOnly: PropTypes.bool,
  // eslint-disable-next-line react/no-unused-prop-types
  removeDocument: PropTypes.func,
  archiveDocuments: PropTypes.func,
  excludedColumnKeys: PropTypes.arrayOf(PropTypes.string),
  additionalColumnProps: PropTypes.arrayOf(
    PropTypes.shape({
      key: PropTypes.string,
      colProps: PropTypes.objectOf(PropTypes.string),
    })
  ),
  // eslint-disable-next-line react/no-unused-prop-types
  documentParent: PropTypes.string,
  documentColumns: PropTypes.arrayOf(PropTypes.object),
  openModal: PropTypes.func.isRequired,
  view: PropTypes.string.isRequired,
};

const defaultProps = {
  documents: [],
  isViewOnly: false,
  removeDocument: () => {},
  archiveDocuments: () => {},
  excludedColumnKeys: [],
  additionalColumnProps: [],
  documentColumns: null,
  documentParent: null,
  view: "standard",
};

const openArchiveModal = (event, props, documents) => {
  event.preventDefault();

  props.openModal({
    props: {
      title: `Archive ${props.documents?.length > 1 ? "Multiple Files" : "File"}`,
      closeModal: props.closeModal,
      handleSubmit: props.archiveDocuments,
      documents,
    },
    content: modalConfig.ARCHIVE_DOCUMENT,
    // width: "75vw",
  });
};

export const DocumentTable = (props) => {
  const isMinimalView = props.view === "minimal";

  let columns = [
    {
      title: "Name",
      key: "name",
      dataIndex: "name",
      sorter: !isMinimalView && nullableStringSorter("name"),
      render: (text, record) => {
        if (isMinimalView) {
          return record.name;
        }

        return (
          <div key={record.key} title="Name">
            <DocumentLink
              documentManagerGuid={record.document_manager_guid}
              documentName={record.name}
              truncateDocumentName={false}
            />
          </div>
        );
      },
    },
    {
      title: "Dated",
      key: "dated",
      dataIndex: "dated",
      sorter: !isMinimalView && dateSorter("dated"),
      defaultSortOrder: "descend",
      render: (text) => <div title="Dated">{formatDate(text)}</div>,
    },
    {
      title: "Category",
      key: "category",
      dataIndex: "category",
      sorter: !isMinimalView && nullableStringSorter("category"),
      render: (text) => <div title="Category">{text}</div>,
    },
    {
      title: "Uploaded",
      key: "uploaded",
      dataIndex: "uploaded",
      sorter: !isMinimalView && dateSorter("uploaded"),
      defaultSortOrder: "descend",
      render: (text) => <div title="Uploaded">{formatDate(text)}</div>,
    },
    {
      key: "remove",
      className: props.isViewOnly || !props.removeDocument ? "column-hide" : "",
      render: (text, record) => (
        <div className={props.isViewOnly || !props.removeDocument ? "column-hide" : ""}>
          <Popconfirm
            placement="topLeft"
            title={`Are you sure you want to delete ${record.name}?`}
            onConfirm={(event) => props.removeDocument(event, record.key, props?.documentParent)}
            okText="Delete"
            cancelText="Cancel"
          >
            <Button ghost type="primary" size="small">
              <img src={TRASHCAN} alt="Remove" />
            </Button>
          </Popconfirm>
        </div>
      ),
    },
    {
      key: "archive",
      className: props.isViewOnly || !props.archiveDocuments ? "column-hide" : "",
      render: (text, record) => (
        <div className={props.isViewOnly || !props.archiveDocuments ? "column-hide" : ""}>
          <Button
            ghost
            type="primary"
            size="small"
            onClick={(event) => openArchiveModal(event, props, [record])}
          >
            Archive
          </Button>
        </div>
      ),
    },
  ];

  if (isMinimalView) {
    props.excludedColumnKeys = props.excludedColumnKeys.concat(["remove", "archive", "category"]);
  }

  if (!some(props.documents, "dated")) {
    columns = columns.filter((column) => column.key !== "dated");
  }

  if (props?.excludedColumnKeys?.length > 0) {
    columns = columns.filter((column) => !props.excludedColumnKeys.includes(column.key));
  }

  if (props?.additionalColumnProps?.length > 0) {
    // eslint-disable-next-line no-unused-expressions
    props?.additionalColumnProps.forEach((addColumn) => {
      const columnIndex = columns.findIndex((column) => addColumn?.key === column.key);
      if (columnIndex >= 0) {
        columns[columnIndex] = { ...columns[columnIndex], ...addColumn?.colProps };
      }
    });
  }

  return (
    <Table
      pagination={false}
      columns={props?.documentColumns ?? columns}
      locale={{ emptyText: "No Data Yet" }}
      dataSource={props.documents}
      showHeader={!isMinimalView}
      size={isMinimalView ? "small" : undefined}
      rowClassName={isMinimalView ? "ant-table-row-minimal" : undefined}
    />
  );
};

DocumentTable.propTypes = propTypes;
DocumentTable.defaultProps = defaultProps;

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      openModal,
      closeModal,
    },
    dispatch
  );

export default connect(null, mapDispatchToProps)(DocumentTable);
