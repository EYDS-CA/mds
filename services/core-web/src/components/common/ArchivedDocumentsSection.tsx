import React from "react";
import DocumentTable from "./DocumentTable";
import { Typography } from "antd";
import { IMineDocument } from "@mds/common";
import { DeleteOutlined } from "@ant-design/icons";

interface ArchivedDocumentsSectionProps {
  documents: IMineDocument;
}

const ArchivedDocumentsSection = (props: ArchivedDocumentsSectionProps) => {
  const docs = props.documents.map((d) => {
    d.name = d.document_name;

    return d;
  });

  return (
    <div id="archived-documents">
      <Typography.Title level={4}>
        <DeleteOutlined className="violet" />
        &nbsp;Archived Documents{" "}
      </Typography.Title>
      <Typography.Paragraph>
        These files are not reviewed as part of the submission.
      </Typography.Paragraph>
      <DocumentTable documents={docs} excludedColumnKeys={["archive", "remove"]}>
        {" "}
      </DocumentTable>
    </div>
  );
};

export default ArchivedDocumentsSection;
