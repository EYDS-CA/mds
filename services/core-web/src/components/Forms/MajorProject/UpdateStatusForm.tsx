import React, { FC } from "react";
import { Field } from "redux-form";
import { Col, Row } from "antd";
import { required } from "@mds/common/redux/utils/Validate";
import { IOption, MAJOR_MINE_APPLICATION_AND_IRT_STATUS_CODE_CODES, PROJECT_STATUS_CODES } from "@mds/common";
import FormWrapper from "@mds/common/components/forms/FormWrapper";
import RenderSelect from "@mds/common/components/forms/RenderSelect";
import RenderSubmitButton from "@mds/common/components/forms/RenderSubmitButton";
import ProjectCallout from "@mds/common/components/projects/ProjectCallout";

interface UpdateStatusFormProps {
  displayValues: {
    status_code: MAJOR_MINE_APPLICATION_AND_IRT_STATUS_CODE_CODES | PROJECT_STATUS_CODES,
    update_user: string,
    updateDate: string,
  };
  dropdownOptions: IOption[];
  handleSubmit: (values) => Promise<void>;
  formName: string;
}

const UpdateStatusForm: FC<UpdateStatusFormProps> = ({
  displayValues,
  handleSubmit,
  formName,
  dropdownOptions
}) => {

  return (
    <FormWrapper
      name={formName}
      onSubmit={handleSubmit}
      initialValues={displayValues}
      reduxFormConfig={{
        enableReinitialize: true
      }}
    >
      <Row>
        <Col span={24}>
          <ProjectCallout
            status_code={displayValues.status_code}
            formField={<>
              <Field
                id="status_code"
                name="status_code"
                label=""
                placeholder="Action"
                component={RenderSelect}
                validate={[required]}
                data={dropdownOptions}
                allowClear={false}
              />
              <RenderSubmitButton buttonText="Update Status" />
            </>}
          />
        </Col>
      </Row>
    </FormWrapper>
  )
};

export default UpdateStatusForm;