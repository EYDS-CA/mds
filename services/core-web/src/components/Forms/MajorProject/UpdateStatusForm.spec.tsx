import React from "react";
import { render } from "@testing-library/react";
import { AUTHENTICATION } from "@mds/common/constants/reducerTypes";
import { FORM, PROJECT_STATUS_CODES, SystemFlagEnum } from "@mds/common/index";
import * as MOCK from "@mds/common/tests/mocks/dataMocks";
import { ReduxWrapper } from "@mds/common/tests/utils/ReduxWrapper";
import UpdateStatusForm from "./UpdateStatusForm";

const msState = {
    [AUTHENTICATION]: {
        systemFlag: SystemFlagEnum.ms
    }
};
const coreState = {
    [AUTHENTICATION]: {
        systemFlag: SystemFlagEnum.core
    }
};

describe("UpdateStatusForm", () => {
    it("renders properly", () => {
        const { container } = render(
            <>
                <ReduxWrapper initialState={msState}>
                    <p>MineSpace render: </p>
                    <UpdateStatusForm
                        displayValues={{
                            status_code: PROJECT_STATUS_CODES.ASG,
                            update_user: "username",
                            updateDate: "Jun 13 2022"
                        }}
                        handleSubmit={jest.fn()}
                        formName={FORM.UPDATE_IRT}
                        dropdownOptions={MOCK.INFORMATION_REQUIREMENTS_TABLE_STATUS_CODES_DROPDOWN}
                    />
                </ReduxWrapper>
                <ReduxWrapper initialState={coreState}>
                    <p>CORE render: </p>
                    <UpdateStatusForm
                        displayValues={{
                            status_code: PROJECT_STATUS_CODES.ASG,
                            update_user: "username",
                            updateDate: "Jun 13 2022"
                        }}
                        handleSubmit={jest.fn()}
                        formName={FORM.UPDATE_IRT}
                        dropdownOptions={MOCK.INFORMATION_REQUIREMENTS_TABLE_STATUS_CODES_DROPDOWN}
                    />
                </ReduxWrapper>
            </>
        );

        expect(container).toMatchSnapshot();
    });
});