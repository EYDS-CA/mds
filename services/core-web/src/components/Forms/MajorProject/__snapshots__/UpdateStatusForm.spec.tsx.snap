// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateStatusForm renders properly 1`] = `
<div>
  <p>
    MineSpace render: 
  </p>
  <form
    class="ant-form ant-form-vertical common-form common-form-UPDATE_IRT form-edit"
    id="UPDATE_IRT"
  >
    <div
      class="ant-row"
    >
      <div
        class="ant-col ant-col-24"
      >
        <div
          class="bcgov-callout--success bcgov-callout"
        >
          <p
            class="bcgov-callout-title"
          >
            Submitted
          </p>
          <p>
            This project has been formally submitted by the proponent through MineSpace. MineSpace users can update documents only.
          </p>
        </div>
      </div>
    </div>
  </form>
  <p>
    CORE render: 
  </p>
  <form
    class="ant-form ant-form-vertical common-form common-form-UPDATE_IRT form-edit"
    id="UPDATE_IRT"
  >
    <div
      class="ant-row"
    >
      <div
        class="ant-col ant-col-24"
      >
        <div
          class="ant-alert ant-alert-warning ant-alert-with-description margin-large--bottom"
          data-show="true"
          role="alert"
        >
          <span
            aria-label="exclamation-circle"
            class="anticon anticon-exclamation-circle ant-alert-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="exclamation-circle"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
              />
              <path
                d="M464 688a48 48 0 1096 0 48 48 0 10-96 0zm24-112h48c4.4 0 8-3.6 8-8V296c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8z"
              />
            </svg>
          </span>
          <div
            class="ant-alert-content"
          >
            <div
              class="ant-alert-message"
            >
              Assigned
            </div>
            <div
              class="ant-alert-description"
            >
              <div
                class="ant-row ant-row-space-between"
              >
                <div
                  class="ant-col ant-col-xs-24 ant-col-md-18"
                >
                  This project has been formally submitted by the proponent through MineSpace. MineSpace users can update documents only.
                </div>
                <div
                  class="ant-col ant-col-xs-24 ant-col-md-6"
                >
                  <div
                    class="ant-form-item ant-form-item-with-help"
                  >
                    <div
                      class="ant-row ant-form-item-row"
                    >
                      <div
                        class="ant-col ant-form-item-control"
                      >
                        <div
                          class="ant-form-item-control-input"
                        >
                          <div
                            class="ant-form-item-control-input-content"
                          >
                            <div
                              class="ant-select ant-select-in-form-item ant-select-single ant-select-show-arrow ant-select-show-search"
                              data-cy="status_code"
                            >
                              <div
                                class="ant-select-selector"
                              >
                                <span
                                  class="ant-select-selection-search"
                                >
                                  <input
                                    aria-activedescendant="status_code_list_0"
                                    aria-autocomplete="list"
                                    aria-controls="status_code_list"
                                    aria-haspopup="listbox"
                                    aria-owns="status_code_list"
                                    autocomplete="off"
                                    class="ant-select-selection-search-input"
                                    id="status_code"
                                    role="combobox"
                                    type="search"
                                    value=""
                                  />
                                </span>
                                <span
                                  class="ant-select-selection-item"
                                  title="ASG"
                                >
                                  ASG
                                </span>
                              </div>
                              <span
                                aria-hidden="true"
                                class="ant-select-arrow"
                                style="user-select: none;"
                                unselectable="on"
                              >
                                <span
                                  aria-label="down"
                                  class="anticon anticon-down ant-select-suffix"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="down"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </div>
                          </div>
                        </div>
                        <div
                          style="display: flex; flex-wrap: nowrap;"
                        >
                          <div
                            class="ant-form-item-explain ant-form-item-explain-connected"
                            id="UPDATE_IRT_status_code_help"
                            role="alert"
                          >
                            <div
                              class=""
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <button
                    aria-label="Submit"
                    class="ant-btn ant-btn-primary"
                    disabled=""
                    type="submit"
                  >
                    <span>
                      Update Status
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
